sdk/src/main/java/com/hedera/hashgraph/sdk/HederaTrustManager.java

```diff
@@ -0,0 +1,63 @@
+package com.hedera.hashgraph.sdk;
+
+import org.bouncycastle.crypto.digests.SHA384Digest;
+import org.bouncycastle.util.encoders.Hex;
+
+import javax.annotation.Nullable;
+import javax.net.ssl.X509TrustManager;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+class HederaTrustManager implements X509TrustManager {
+    @Nullable
+    private final String certHash;
+
+    HederaTrustManager(@Nullable NetworkName networkName, AccountId nodeAccountId) {
+        if (networkName == null) {
+            this.certHash = null;
+            return;
+        }
+
+        var nodeAddressBook = Network.nodeAddressBooks.get(networkName);
```

Comment:
Passing around `NetworkName` then using that to look into a static hashmap of `Network` is not a very good abstraction and highly couples these all together. Recommend a `Network` instance encapsulate everything and a `network.getAddressBook()` be added. That will also better support non-public networks.